### DISCUSSION TITLE
[Mod Change Proposal] Strict Tracking --> Strict Play

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#18083] [Ideas]
I found the Strict Tracking mod and I absolutely love it! It's a wonderful tool to prevent "cheese" by punishing the player for doing things wrong, whereas the mistake would usually be brushed aside. In my perspective, this is a wonderful tool to assist beginner and average players by treating mistakes with a plane crash mentality as opposed to a car crash one, this was one of the reasons I love the style of valuing accuracy on sliders, when your acc drops to 70% whereas you could get 85% without scoreV2, it highlights the lack of skill drastically.

I would like to simply add on to what exists so that this mod can do the task even better.

---

More strict tracking:

Killing combo for releasing pre-emptively or tracking a slider incorrectly punishes indecisive/incorrect play, to add on to this: ***_1) At the end of each slider, there is a hit timer that the player has to release or have a dropped combo._***

A player can just hold on to a key for a slider however long they want, which is a bad practice for players to get into as they will be looking to play the next circle to the next beat. When circles start to come soon after sliders end, players may often mistake the upcoming circle to be timed at the time that the slider **should** end. I know this from experience.

---

Stricter streaming/bursts:

***2) Tapping a key when the cursor isn't over anything results in a dropped combo***
- also counts for when the cursor is over a note that isn't playable yet
- shouldn't proc if the next note is a spinner

Its so easy for new players that start streaming to just spam their two keys without any regard to the rhythm, one that I catch myself doing from time to time is ignoring skipped beats that happen during jumps, I just stream throughout the jump instead of skipping that beat.

***3) Tapping a key while the other key is still held down results in dropped combo***
- Very well may be unfeasible to implement, could proc unintentionally a lot during fast streaming.

This mainly targets double-tapping, but it also serves as a redundancy for the first change that prevents players from holding on to sliders for too long.

---

There may be other ways this mod could be designed to prevent even more bad habits, but this is all I've got for now.

It'd be cool to get the option to customize this mod to choose what settings you want active, in case there are some a player didn't want, which had a secondary icon to indicate that while it's not the mod at 100% power, it's still a more restricted mod. This mod is for show, as opposed to one that actively increases PP, since the mod doesn't change how the game is fundamentally player, like sudden death, so there's no need for it to not have customizability.

**Comment by NotGumballer91** [comment]
Some of the things you've suggested seems a bit too strict.
> 1) At the end of each slider, there is a hit timer that the player has to release or have a dropped combo.

I'm not too sure. There's no slider velocity indicator and you can't see how fast or slow can the slider can go.
Following **ONLY the sliderball** is strict enough

> 2) Tapping a key when the cursor isn't over anything results in a dropped combo

This is only suggestion that I partially agree. When there's a 3 note burst and the player clicks 4 instead, It results in a miss.
The problem is, It's a like a reverse of notelock. Will it add new miss (even though you could have 100 misses but the map has 50 objects) or automatically miss the next circle when you misclick or "overclick"?

> 3) Tapping a key while the other key is still held down results in dropped combo

Absolutely not. This makes certain stream/technical maps tremendously hard or outright impossible if it's timed tightly or high BPM

**Reply by ghost** [reply]
1. At the end of each slider, there is a hit timer that the player has to release or have a dropped combo.

This was a problem I really struggled with when I was first learning OSU!, I wouldn't see the slider-ends as a part of a beat and it would muddy up my rhythm by overlapping it with the upcoming notes, I actually had to "VOD review" my OSU! gameplay to figure out I was even doing that. As I said, the way I see the mod is as a way to prevent beginners from creating bad habits, so I figured this would work fine, it's even a mechanic in some other rhythm games.

2. Tapping a key when the cursor isn't over anything results in a dropped combo

I didn't elaborate on how exactly it would occur so yes with the former option: _add new miss (even though you could have 100 misses but the map has 50 objects)_. I didn't want to miss the upcoming note but to just miss when you don't have a note to click. As you said, making it possible to have more misses than there are objects in the map.

3. Tapping a key while the other key is still held down results in dropped combo

That's exactly what I was thinking of when I said that it may be unfeasible. While it may not unintentionally affect low-level play, it was possible to make high-level play insufferable.
